app/org/sagebionetworks/bridge/upload/IosSchemaValidationHandler2.java

```diff
@@ -165,41 +180,56 @@ public void handle(@Nonnull UploadValidationContext context)
-        }
-        String item = itemNode.textValue();
-        if (StringUtils.isBlank(item)) {
-            throw new UploadValidationException("info.json \"item\" field is blank");
+        // get relevant params from info.json
+        String item = JsonUtils.asText(infoJson, KEY_ITEM);
+        Integer schemaRev = JsonUtils.asInt(infoJson, KEY_SCHEMA_REV);
+        String surveyGuid = JsonUtils.asText(infoJson, KEY_SURVEY_GUID);
+        String surveyCreatedOn = JsonUtils.asText(infoJson, KEY_SURVEY_CREATED_ON);
+
+        // Old versions of YML apps sometimes send "identifier" instead of "item". This isn't officially supported,
+        // but we have to do it anyway for backwards compatibility.
+        String identifier = JsonUtils.asText(infoJson, KEY_IDENTIFIER);
+
+        if (StringUtils.isNotBlank(surveyGuid) && StringUtils.isNotBlank(surveyCreatedOn)) {
+            // First try getting the schema by survey.
+            return getUploadSchemaBySurvey(study, surveyGuid, surveyCreatedOn);
+        } else if (StringUtils.isNotBlank(item)) {
+            // Fall back to item field.
+            return getUploadSchemaByItemAndRev(study, item, schemaRev);
+        } else if (StringUtils.isNotBlank(identifier)) {
+            // Fall back to identifier field. Log a warning that we're using this non-standard field.
+            logger.warn("info.json missing item field, falling back to identifier field, identifier=" + identifier);
+            return getUploadSchemaByItemAndRev(study, identifier, schemaRev);
+        } else {
+            throw new UploadValidationException(
+                    "info.json must contain either item or surveyGuid and surveyCreatedOn");
         }
+    }
 
-        // extract schema rev
-        Integer schemaRev = null;
-        JsonNode schemaRevNode = infoJson.get(KEY_SCHEMA_REV);
-        if (schemaRevNode != null) {
-            if (schemaRevNode.isNull()) {
-                throw new UploadValidationException("info.json has null \"schemaRevision\" field");
-            }
-            if (!schemaRevNode.isIntegralNumber()) {
-                throw new UploadValidationException("info.json \"schemaRevision\" is not an integer: "
-                        + schemaRevNode.toString());
-            }
-            schemaRev = schemaRevNode.intValue();
+    private UploadSchema getUploadSchemaBySurvey(StudyIdentifier study, String surveyGuid, String surveyCreatedOn)
+            throws UploadValidationException {
+        // surveyCreatedOn is a timestamp. SurveyService takes long epoch millis. Convert.
+        long surveyCreatedOnMillis= DateUtils.convertToMillisFromEpoch(surveyCreatedOn);
+
+        // Get survey. We use the survey identifier as the schema ID and the schema revision. Both of these must be
+        // specified.
+        Survey survey = surveyService.getSurvey(new GuidCreatedOnVersionHolderImpl(surveyGuid, surveyCreatedOnMillis));
```

Comment:
If this doesn't retrieve a survey it throws an EntityNotFoundException, do you want to translate to a UVE?